retry.go

```diff
@@ -69,14 +70,13 @@ func (c Client) newRetryTimer(maxRetry int, unit time.Duration, cap time.Duratio
 	go func() {
 		defer close(attemptCh)
 		for i := 0; i < maxRetry; i++ {
+			attemptCh <- i + 1
+
```

Comment:
attemptCh should be kept as part of select otherwise your doneCh is never triggered..

Reply:
I remove  doneCh because it not needer any more. Context is uses for breaking retries now.

Reply:
No the correct code should be to keep the attemptCh as part of the select @fortyanov - it won't work as expected for ctx.Done() is called.

Reply:
It should be 

```
select {
   case attemptCh <- i+1:
   case <-time.After(exponentialBackoffWait(i)):
   case ctx.Done():
            return
}
```

Reply:
I made this changes.